streams/src/main/java/org/apache/kafka/streams/tools/KafkaStreamsCleanupClient.java

```diff
@@ -0,0 +1,249 @@
+                .withRequiredArg()
+                .ofType(String.class);
+        window_option = optionParser.accepts("windows", "Comma separted list of window names")
+                .withRequiredArg()
+                .ofType(String.class);
+        state_dir_option = optionParser.accepts("state-dir", "Local state store directory (state.dir)")
+                .withRequiredArg()
+                .defaultsTo("/tmp/kafka-streams")
+                .ofType(String.class);
+
+        try {
+            options = optionParser.parse(args);
+        } catch(OptionException e) {
+            System.err.println(e.getMessage());
+            optionParser.printHelpOn(System.err);
+            System.exit(-1);
+        }
+    }
+
+    private void resetSourceTopicOffsets() {
+        List<String> topics = new LinkedList< >();
+        topics.addAll(options.valuesOf(source_topics_option));
+        topics.addAll(options.valuesOf(intermediate_topics_option));
+
+        if(topics.size() == 0) {
+            System.out.println("No source or intermediate topics specified.");
+            return;
+        }
+
+        System.out.println("Resetting offests to zero for topics " + topics);
+
+        Properties config = new Properties();
+        config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, options.valueOf(bootstrap_server_option));
+        config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(application_id_option));
+        config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+        KafkaConsumer<byte[],byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+        try {
+            client.subscribe(topics);
+            client.poll(1);
+
+            for (TopicPartition partition : client.assignment()) {
+                client.seek(partition, 0);
+            }
+            client.commitSync();
+        } catch(RuntimeException e) {
+            System.err.println("Resetting source topic offsets failed");
+            throw e;
+        } finally {
+            client.close();
+        }
+    }
+
+    private void clearIntermediateTopics(ZkUtils zkUtils, List<String> allTopics) {
+        if (options.has(intermediate_topics_option)) {
+            System.out.println("Clearing intermediate user topics.");
+
+            HashMap<String, String> topicRetentionTime = new HashMap<>();
+            for (String topic : options.valuesOf(intermediate_topics_option)) {
+                if (allTopics.contains(topic)) {
+                    Properties configs = AdminUtils.fetchEntityConfig(zkUtils, "topics", topic);
+                    topicRetentionTime.put(topic, configs.getProperty("retention.ms"));
+
+                    configs.setProperty("retention.ms", "1");
+                    AdminUtils.changeTopicConfig(zkUtils, topic, configs);
+                }
+            }
+
+            // TODO fix -- simply sleep is not good enough
```

Comment:
I believe we can do this by submitting OffsetRequests for the topics, i.e., get the earliest and latest offsets for each partition and when the difference is 0 then we know we have 0 messages in the partition